### PR TITLE
Alpakka 2.0.1 update, API changes, some minor fixes.

### PR DESCRIPTION
**Pull request by RayRoestenburg**
### What changes were proposed in this pull request?

- Changes to API in AkkaStreamlet, update of Alpakka Kafka 2.0.1

### Why are the changes needed?

- Update of Alpakka Kafka 2.0.1

### CODE REVIEW

File: shared-content-source/docs/modules/develop/pages/build-akka-streamlets.adoc [hunk 1]
```diff
@@ -382,18 +382,27 @@ The `plainSource` method optionally takes a `ResetPosition` argument. By default
...
 ====
 
-`StreamletLogic` also provides a `Source` that makes it possible to process records at-least-once. 
-The `sourceWithOffsetContext` method creates a `Source`, where every record is automatically associated with a committable offset. 
-That committable offset must be committed once you are done processing. 
+`StreamletLogic` also provides a special `Source`, a https://doc.akka.io/api/akka/current/akka/stream/scaladsl/SourceWithContext.html[Scala SourceWithContext] / https://doc.akka.io/japi/akka/current/akka/stream/scaladsl/SourceWithContext.html[Java SourceWithContext], that makes it possible to process records at-least-once. 
+The `sourceWithOffsetContext` method creates a `SourceWithContext`, where every record is automatically associated with a `CommittableOffset`. (the context for every message is a `CommittableOffset`)
 
-There are several ways to commit offsets:
+The `CommittableOffset` extends `Committable`, which is a token that can be used to indicate that you have read and processed the associated message, and do not want to process it again when the streamlet restarts.
```
Comment (olofwalker) [inline shared-content-source/docs/modules/develop/pages/build-akka-streamlets.adoc]: ` which is a token that can be used `

This sounds very cryptic to me, what token?

File: shared-content-source/docs/modules/develop/pages/build-akka-streamlets.adoc [hunk 2]
```diff
@@ -382,18 +382,27 @@ The `plainSource` method optionally takes a `ResetPosition` argument. By default
 If `ResetPosition` is set to `Earliest`, `plainSource` will start providing records from the earliest offset known for every partition, potentially re-reading *all received records* on *every* restart.
 ====
 
-`StreamletLogic` also provides a `Source` that makes it possible to process records at-least-once. 
-The `sourceWithOffsetContext` method creates a `Source`, where every record is automatically associated with a committable offset. 
-That committable offset must be committed once you are done processing. 
+`StreamletLogic` also provides a special `Source`, a https://doc.akka.io/api/akka/current/akka/stream/scaladsl/SourceWithContext.html[Scala SourceWithContext] / https://doc.akka.io/japi/akka/current/akka/stream/scaladsl/SourceWithContext.html[Java SourceWithContext], that makes it possible to process records at-least-once.
```
Comment (olofwalker) [inline shared-content-source/docs/modules/develop/pages/build-akka-streamlets.adoc]: Why is it special?

File: shared-content-source/docs/modules/develop/pages/use-akka-streamlets.adoc [hunk 3]
```diff
@@ -82,10 +82,10 @@ At-most-once message delivery semantics are applied to Akka streamlets that use
 Akka Streamlets also supports _at-least-once_ semantics for cases where the above mentioned options are not viable.
 In this case, offsets of incoming records are tracked, so that the streamlet can start from a specific offset, per inlet, on restart.
 Once records are written to outlets, the associated incoming record offsets are considered.
-Offsets are committed in batch, and only when the offset changes.
+Offsets are committed in batch.
```
Comment (olofwalker) [inline shared-content-source/docs/modules/develop/pages/use-akka-streamlets.adoc]: Statements like this makes me think "why?" , "for performance reasons" ?

File: shared-content-source/docs/modules/develop/pages/use-akka-streamlets.adoc [hunk 4]
```diff
@@ -82,10 +82,10 @@ At-most-once message delivery semantics are applied to Akka streamlets that use
...
 
-Akka streamlets that want to support at-least-once semantics, must use an Akka Stream `FlowWithContext` that propagates `CommittableOffset`{empty}s through the flow.
-Cloudflow automatically propagates offset details through the flow and commits offset changes in batches.
+Akka streamlets that want to support at-least-once semantics, must use an Akka Stream `FlowWithContext` that propagates `Committable`{empty}s (both offsets and batches of offsets) through the flow.
+Cloudflow automatically propagates `Committable`{empty}s (offset details) through the flow and commits these in batches, acknowledging processed messages. Batches that have been committed will not be read again by the streamlet after a restart.
```
Comment (olofwalker) [inline shared-content-source/docs/modules/develop/pages/use-akka-streamlets.adoc]:
```suggestion
Cloudflow automatically propagates `Committable`{empty}s (offset details) through the flow and commits these in batches, acknowledging processed messages. Committed batches are not reread after a restart.
```

File: shared-content-source/docs/modules/develop/pages/build-akka-streamlets.adoc (hunk 1 above)
Comment (RayRoestenburg) [reply shared-content-source/docs/modules/develop/pages/build-akka-streamlets.adoc]: I'll try to improve the sentence.

File: shared-content-source/docs/modules/develop/pages/build-akka-streamlets.adoc [hunk 5]
```diff
@@ -382,18 +382,30 @@ The `plainSource` method optionally takes a `ResetPosition` argument. By default
...
+`StreamletLogic` also provides a source type that can pass along context, see https://doc.akka.io/api/akka/current/akka/stream/scaladsl/SourceWithContext.html[Scaladocs SourceWithContext] / https://doc.akka.io/japi/akka/current/akka/stream/scaladsl/SourceWithContext.html[Javadocs SourceWithContext]. It is used to pass along offsets to the end of the stream, where they can be marked as read, after the associated messages have been processed.
 
-There are several ways to commit offsets:
+The `sourceWithOffsetContext` method creates a `SourceWithContext`, where every record is automatically associated with a `CommittableOffset`. 
 
-`SourceWithContext`:: used to pass through the committable offsets further downstream. 
+Anything that can be marked as read in the stream, to prevent re-processing at restart, extends `Committable`. `CommittableOffset` and `CommittableOffsetBatch` are examples of `Committable` objects. 
 
-`Flow`{empty}s and `Sink`{empty}s:: used to make sure that offsets are committed automatically.
+which is a token that can be used to indicate that you have read and processed the associated message, and do not want to process it again when the streamlet restarts.
```
Comment (rstento) [inline shared-content-source/docs/modules/develop/pages/build-akka-streamlets.adoc]: Is part of this sentence missing? Perhaps something like, The 'Commitable' objects provide tokens ...?

File: shared-content-source/docs/modules/develop/pages/build-akka-streamlets.adoc [hunk 6]
```diff
@@ -382,18 +382,30 @@ The `plainSource` method optionally takes a `ResetPosition` argument. By default
...
 
-All of these types move the committable offset along with the data but allow you to apply normal operators to the stream that treat the data transparently, as if it were without context. 
+Acknowledging the read messages is done by writing the `Committable` to a `committableSink`, which we'll show the details of a little later.
```
Comment (rstento) [inline shared-content-source/docs/modules/develop/pages/build-akka-streamlets.adoc]: Better split into two: Acknowledging the read messages is done by writing the `Committable` to a `committableSink`. We'll show the details later. (or See the xxx section for an example)

File: shared-content-source/docs/modules/develop/pages/build-akka-streamlets.adoc [hunk 7]
```diff
@@ -382,18 +382,30 @@ The `plainSource` method optionally takes a `ResetPosition` argument. By default
...
 
-`Flow`{empty}s and `Sink`{empty}s:: used to make sure that offsets are committed automatically.
+which is a token that can be used to indicate that you have read and processed the associated message, and do not want to process it again when the streamlet restarts.
 
-All of these types move the committable offset along with the data but allow you to apply normal operators to the stream that treat the data transparently, as if it were without context. 
+Acknowledging the read messages is done by writing the `Committable` to a `committableSink`, which we'll show the details of a little later. 
 
+[NOTE]
+====
+Many readers will likely think of _commits_ as ensuring database writes to occur in a single transaction. It's important to note that in the context of Akka streamlets, committing messages means something else entirely. It does not refer to ACID transactions, and does not refer to writing or producing records to the pub/sub system. In this case committing means that the streamlet instance indicates that it has read up to a particular message and will not want to process the message again on restart.
```
Comment (rstento) [inline shared-content-source/docs/modules/develop/pages/build-akka-streamlets.adoc]: To make even more clear, I'd replace "In this case" with "Rather, in Akka streamlets, committing means ...

File: shared-content-source/docs/modules/develop/pages/build-akka-streamlets.adoc (hunk 5 above)
Comment (RayRoestenburg) [reply shared-content-source/docs/modules/develop/pages/build-akka-streamlets.adoc]: This was a leftover, good catch!

**Pull request merged**